src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java

```diff
@@ -140,16 +165,199 @@ private int calculateTotal(int base, TFloatList multipliers, TIntList modifiers)
+     *
+     * @param event OnDamagedEvent triggered when entity is damaged.
+     * @param entity Entity which is damaged.
+     * @param characterSounds Component having sound settings.
+     */
+    @ReceiveEvent
+    public void onDamaged(OnDamagedEvent event, EntityRef entity, CharacterSoundComponent characterSounds) {
+        if (characterSounds.lastSoundTime + CharacterSoundSystem.MIN_TIME < time.getGameTimeInMs()) {
+
+            // play the sound of damage hitting the character for everyone
+            DamageSoundComponent damageSounds = event.getType().getComponent(DamageSoundComponent.class);
+            if (damageSounds != null && !damageSounds.sounds.isEmpty()) {
+                StaticSound sound = random.nextItem(damageSounds.sounds);
+                if (sound != null) {
+                    entity.send(new PlaySoundEvent(sound, 1f));
+                }
+            }
+
+            // play the sound of a client's character being damaged to the client
+            if (!characterSounds.damageSounds.isEmpty()) {
+                StaticSound sound = random.nextItem(characterSounds.damageSounds);
+                if (sound != null) {
+                    entity.send(new PlaySoundForOwnerEvent(sound, characterSounds.damageVolume));
+                }
+            }
+
+            characterSounds.lastSoundTime = time.getGameTimeInMs();
+            entity.saveComponent(characterSounds);
+
+        }
+    }
+
+    /**
+     * Causes damage to entity when fallingDamageSpeedThreshold is breached.
+     *
+     * @param event VerticalCollisionEvent sent when falling speed threshold is crossed.
+     * @param entity The entity which is damaged due to falling.
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onLand(VerticalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+        float speed = Math.abs(event.getVelocity().y);
+
+        if (speed > health.fallingDamageSpeedThreshold) {
+            int damage = (int) ((speed - health.fallingDamageSpeedThreshold) * health.excessSpeedDamageMultiplier);
+            if (damage > 0) {
+                checkDamage(entity, damage, EngineDamageTypes.PHYSICAL.get(), EntityRef.NULL, EntityRef.NULL);
+            }
+        }
+    }
+
+    /**
+     * Inflicts damage to entity horizontalDamageSpeedThreshold is breached.
+     *
+     * @param event HorizontalCollisionEvent sent when "falling horizontally".
+     * @param entity Entity which is damaged on "horizontal fall".
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onCrash(HorizontalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+
+        Vector3f vel = new Vector3f(event.getVelocity());
+        vel.y = 0;
+        float speed = vel.length();
+
+        if (speed > health.horizontalDamageSpeedThreshold) {
```

Comment:
Parts of this method and the `onLand` method could likely be extracted into a single `highSpeedDamage` method